### PR TITLE
Zyre Leader Election

### PR DESCRIPTION
**Pull request by sappo**
This PR adds leader election to Zyre. This introduces two new messages to Zyre protocol.

I'm in progress of updating the RFC.

### CODE REVIEW

**Comment by bluca** [comment]
Nice!

Just 2 small notes: the private API is private and hence doesn't need to be marked as stable/draft - it won't be visible outside
Also the new RFC for Zyre is https://rfc.zeromq.org/spec:43/ZRE/

**Comment by sappo** [comment]
All right thanks for the advise.

**Comment by bluca** [comment]
Looks like the linker is not happy:

```
zyre_election.c:(.text+0xb9e): undefined reference to `zyre_set_contest_in_group'
```

File: src/zyre.c [hunk 1]
```diff
@@ -277,6 +277,23 @@ zyre_set_endpoint (zyre_t *self, const char *format, ...)
     return zsock_wait (self->actor) == 0? 0: -1;
 }
 
+#ifdef ZYRE_BUILD_DRAFT_API
```
Comment (bluca) [inline src/zyre.c]: @sappo this is causing the linking error - the draft/nondraft is in the headers, and affects the visibility attribute used with the declarations, but in the body we always define the functions

File: src/zyre_election.h [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+//  Enable/disable verbose logging.
+ZYRE_PRIVATE void
+    zyre_election_set_verbose (zyre_election_t *self, bool verbose);
+
+//  Print election status to command line
+ZYRE_PRIVATE void
+    zyre_election_print (zyre_election_t *self);
+
+//  Self test of this class
+ZYRE_EXPORT void
```
Comment (bluca) [inline src/zyre_election.h]: @sappo this should be _PRIVATE, the compiler warns of it:
```
In file included from src/zyre_node.c:16:0:
src/zyre_classes.h:116:5: warning: redeclaration of 'zyre_election_test' with different visibility (old visibility preserved) [enabled by default]
     zyre_election_test (bool verbose);
     ^
In file included from src/zyre_classes.h:57:0,
                 from src/zyre_node.c:16:
src/zyre_election.h:104:5: note: previous declaration of 'zyre_election_test' was here
     zyre_election_test (bool verbose);
     ^
```

**Comment by bluca** [comment]
Nice stuff!

**Pull request merged**